<details><summary>Linter check found the following problems:</summary>
Traceback (most recent call last):
  File "/opt/conda/envs/linter/lib/python3.13/site-packages/anaconda_linter/run.py", line 208, in main
    return_code, report = execute_linter(
                          ~~~~~~~~~~~~~~^
        recipe=args.recipe,
        ^^^^^^^^^^^^^^^^^^^
    ...<6 lines>...
        verbose_flag=args.verbose,
        ^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/linter/lib/python3.13/site-packages/anaconda_linter/run.py", line 180, in execute_linter
    result = linter.lint(recipes, subdir, variant_config_files, exclusive_config_files, fix_flag)
  File "/opt/conda/envs/linter/lib/python3.13/site-packages/anaconda_linter/lint/__init__.py", line 725, in lint
    msgs = self.lint_file(
        recipe_name,
    ...<3 lines>...
        fix=fix,
    )
  File "/opt/conda/envs/linter/lib/python3.13/site-packages/anaconda_linter/lint/__init__.py", line 831, in lint_file
    self.lint_recipe(
    ~~~~~~~~~~~~~~~~^
        recipe,
        ^^^^^^^
        fix=fix,
        ^^^^^^^^
    )
    ^
  File "/opt/conda/envs/linter/lib/python3.13/site-packages/anaconda_linter/lint/__init__.py", line 891, in lint_recipe
    res = self.check_instances[check].run(recipe, fix)
  File "/opt/conda/envs/linter/lib/python3.13/site-packages/anaconda_linter/lint/__init__.py", line 294, in run
    self.check_recipe(recipe)
    ~~~~~~~~~~~~~~~~~^^^^^^^^
  File "/opt/conda/envs/linter/lib/python3.13/site-packages/anaconda_linter/lint/check_syntax.py", line 36, in check_recipe
    has_constraints = constraints.search(spec)
TypeError: expected string or bytes-like object, got 'NoneType'

ERROR conda.cli.main_run:execute(127): `conda run conda-lint /tmp/abs_a0uj07ordd/clone` failed. (See above for error)
</details>

--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -85,11 +85,21 @@ meson setup builddir/ \
   -Dlibunwind=enabled \
   -Dshared-glapi=enabled \
   -Dxmlconfig=enabled \
+  -Dglvnd-vendor-name=${glvnd_vendor_name} \
   || { cat builddir/meson-logs/meson-log.txt; exit 1; }
 
 ninja -C builddir/ -j ${CPU_COUNT}
 
 ninja -C builddir/ install
+
+# Only need this on Linux since OSX doesn't make EGL or GLX.
+if [[ "$target_platform" == linux* ]]; then
+  # To make sure our EGL mesa libs don't get picked up on accident we use an alternate GLVND name and ICD with
+  # a higher priority default file in the same location. Just renaming the GLX ones is enough.
+  cp ${PREFIX}/share/glvnd/egl_vendor.d/50_${glvnd_vendor_name}.json ${PREFIX}/share/glvnd/egl_vendor.d/99_${glvnd_vendor_name}.json
+  mv ${PREFIX}/share/glvnd/egl_vendor.d/50_${glvnd_vendor_name}.json ${PREFIX}/share/glvnd/egl_vendor.d/50_mesa.json
+  sed -i "s:libEGL_${glvnd_vendor_name}.so:libEGL_mesa.so:g" ${PREFIX}/share/glvnd/egl_vendor.d/50_mesa.json
+fi
 
 echo "Running Mesa tests..."
 meson test -C builddir/ -t 4 -v

--- a/recipe/conda_build_config.yaml
+++ b/recipe/conda_build_config.yaml
@@ -1,2 +1,1 @@
-MACOSX_SDK_VERSION:            # [osx and x86_64]
-  - '10.15'                    # [osx and x86_64]
+glvnd_vendor_name: "anaconda_mesa"

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -1,4 +1,4 @@
-{% set version = "25.1.1" %}
+{% set version = "25.1.5" %}
 
 package:
   name: mesalib
@@ -6,7 +6,7 @@ package:
 
 source:
   url: https://archive.mesa3d.org/mesa-{{ version }}.tar.xz
-  sha256: cf942a18b7b9e9b88524dcbf0b31fed3cde18e6d52b3375b0ab6587a14415bce
+  sha256: 3c4f6b10ff6ee950d0ec6ea733cc6e6d34c569454e3d39a9b276de9115a3b363
 
 build:
   number: 0
@@ -20,6 +20,7 @@ build:
 
 requirements:
   build:
+    - {{ stdlib('c') }}
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
     - meson >=0.63.0
@@ -70,16 +71,19 @@ test:
   commands:
     # We assert that we didn't build hardware accelerated screen rendering capabilities.
     # We want on-screen rendering capabilities to be hardware accelerated but that will take more effort.
-    - test -f $PREFIX/lib/dri/kms_swrast_dri${SHLIB_EXT}        # [linux]
-    - test -f $PREFIX/lib/dri/swrast_dri${SHLIB_EXT}            # [linux]
-    - test -f $PREFIX/lib/libgbm${SHLIB_EXT}                    # [linux]
-    - test -f $PREFIX/lib/libGLX_mesa${SHLIB_EXT}               # [linux]
-    - test -f $PREFIX/lib/libEGL_mesa${SHLIB_EXT}               # [linux]
-    - test -f $PREFIX/lib/libvulkan_lvp${SHLIB_EXT}             # [unix]
-    - test -f $PREFIX/lib/gallium-pipe/pipe_swrast${SHLIB_EXT}  # [unix]
-    - test -f $PREFIX/share/vulkan/icd.d/lvp_icd.x86_64.json    # [x86_64]
-    - test -f $PREFIX/share/vulkan/icd.d/lvp_icd.aarch64.json   # [aarch64]
-    - test -f $PREFIX/share/vulkan/icd.d/lvp_icd.ppc64le.json   # [ppc64le]
+    - test -f $PREFIX/lib/dri/kms_swrast_dri${SHLIB_EXT}              # [linux]
+    - test -f $PREFIX/lib/dri/swrast_dri${SHLIB_EXT}                  # [linux]
+    - test -f $PREFIX/lib/libgbm${SHLIB_EXT}                          # [linux]
+    - test -f $PREFIX/lib/libGLX_{{ glvnd_vendor_name }}${SHLIB_EXT}  # [linux]
+    - test -f $PREFIX/lib/libEGL_{{ glvnd_vendor_name }}${SHLIB_EXT}  # [linux]
+    - test -f $PREFIX/lib/libvulkan_lvp${SHLIB_EXT}                   # [unix]
+    - test -f $PREFIX/lib/gallium-pipe/pipe_swrast${SHLIB_EXT}        # [unix]
+    - test -f $PREFIX/share/vulkan/icd.d/lvp_icd.x86_64.json          # [x86_64]
+    - test -f $PREFIX/share/vulkan/icd.d/lvp_icd.aarch64.json         # [aarch64]
+    - test -f $PREFIX/share/vulkan/icd.d/lvp_icd.ppc64le.json         # [ppc64le]
+
+    - test -f $PREFIX/share/glvnd/egl_vendor.d/99_{{ glvnd_vendor_name }}.json  # [linux]
+    - test -f $PREFIX/share/glvnd/egl_vendor.d/50_mesa.json                     # [linux]
 
     - test ! -f $PREFIX/lib/libGLESv1_CM.1${SHLIB_EXT}  # [osx]
     - test ! -f $PREFIX/lib/libGLESv2.2${SHLIB_EXT}     # [osx]

